app/portainer/services/api/tagService.js

```diff
@@ -35,12 +35,13 @@ angular.module('portainer.app')
     return deferred.promise;
   };
 
-  service.createTag = function(name) {
+  service.createTag = async function(name) {
     var payload = {
       Name: name
     };
 
-    return Tags.create({}, payload).$promise;
+    const tag = await Tags.create({}, payload).$promise;
+    return new TagViewModel(tag);
```

Comment:
Protect async with try/catch closure and throw a custom error : `new PortainerError("Unable to create tag", error)` or `{msg: "Unable to create tag", err: error}`

Reply:
didn't find PortainerError in our codebase, and `new Error` gets only a string

Reply:
throw just an object without `new Error` around, will discard the stack trace

Reply:
PortainerError has been introduced in k8s codebase, sorry.
Throwing a plain object with `msg` and `err` fields will make it suitable for our Notification service.